deepspeed/checkpoint/constants.py

```diff
@@ -74,10 +74,14 @@
 # Similarly, load_hp_checkpoint_state has to take the needed actions when loading from universal.
 PARAM_N_SUB_PARAMS = "param_n_sub_params"
 
+SUB_PARAM_SHAPE = "sub_param_shape"
+
 # Regex list of parameters that require special handling
 VOCABULARY_PARAMETER_PATTERNS = 'vocabulary_parameter_patterns'
 PIPELINE_REPLICATED_PARAMETER_PATTERNS = 'pipeline_replicated_parameter_patterns'
 PARAMETER_TO_AVERAGE_PATTERNS = 'parameter_to_average_patterns'
 PARAMETER_WITH_ROW_PARALLELISM_PATTERNS = 'parameter_with_row_parallelism_patterns'
 TP_REPLICATED_PARAMETER_PATTERNS = 'tp_replicated_parameter_patterns'
 PARAMETER_WITH_2_SUB_PARAMS_CAT_DIM_0 = 'parameter_with_2_sub_params_cat_dim_0'
+PARAMETER_WITH_SUB_PARAMS = 'parameter_with_sub_params'
```

Comment:
@tohtana, I really like these new constructs: `PARAMETER_WITH_SUB_PARAMS` and `PARAMETER_WITH_SUB_PARAMS`. It seems to me that they can capture a broad range of patterns. For example, it seems they can represent the following: 
1. `PARAMETER_WITH_ROW_PARALLELISM_PATTERNS`,
2. `PARAMETER_WITH_2_SUB_PARAMS_CAT_DIM_0`

Am I correct?